src/main/java/core/basesyntax/storage/impl/StorageImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.storage.impl;
+
+import core.basesyntax.storage.Storage;
+import java.util.Collection;
+import java.util.HashMap;
+
+public class StorageImpl implements Storage {
+    private final HashMap<String, StorageEntry> entries;
+
+    public StorageImpl(HashMap<String, StorageEntry> entries) {
+        this.entries = entries;
+    }
+
+    public StorageImpl() {
+        this.entries = new HashMap<>();
+    }
+
+    @Override
+    public Collection<StorageEntry> getAllEntries() {
+        return this.entries.values();
+    }
+
+    @Override
+    public void updateEntry(String product, int quantity) {
```

Comment:
The method 'updateEntry' should check the resulting balance before saving it in the Storage to ensure it is not negative, as per the checklist requirement for handling the Purchase operation.